src/Normalizer/ErrorNormalizer.php

```diff
@@ -17,6 +17,7 @@ class ErrorNormalizer implements NormalizerInterface, TypeAwareInterface
      */
     public function normalize($result, NormalizationContext $normalizationContext)
     {
+        $normalizationContext->markNullValuesForRemoval();
```

Comment:
Don't see the actual need for this, apart from being more aesthetic.

This way you loose fixed structure, in other languages, you'll have to check additionally for whether or not a property is defined. Of course, you may argue that in JS you can use `optional chaining` (which is still proposal) or even use `lodash`:
```javascript
error?.errors?.map(error => <ErrorBlock error={error} />);
get(error, 'errors', []).map(...);
```
But I think this adds more cognitive load when compared to plain:
```javascript
error.errors.map(error => <ErrorBlock error={error} />);
```

I would suggest even to change the line `27` to always return an array or having this option configurable.

Reply:
Not returning `null` where `array` is expected is actually the main point here. But always returning `errors` even with simple error codes seems a bit strange, too.. For example:
```
{"error": "limit_violation","errors":[]}
```

This field is only used for validation-based errors with translated items for each of the fields.

Reply:
You look from the resource perspective, I see it from the programmer's.
It's easier when you don't have to check for mixed types: check whether it's defined, then check if it's an array, only then iterate over it.